lib/meadow/data/schemas/batch.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Meadow.Data.Schemas.Batch do
+  @moduledoc """
+  Batch edit (update/delete) schema
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @statuses ~w(queued in_progress complete error)
+  @types ~w(update delete)
+
+  @primary_key {:id, Ecto.UUID, autogenerate: false, read_after_writes: true}
+  schema "batches" do
+    field :nickname, :string
+    field :status, :string, default: "queued"
+    field :user, :string
+    field :started, :utc_datetime_usec
+    field :type, :string
+    field :works_updated, :integer
+    field :query, :string
+    field :add, :string
+    field :delete, :string
+    field :replace, :string
+    field :error, :string
+    field :active, :boolean, default: false
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(controlled_value, attrs) do
+    controlled_value
+    |> cast(attrs, [
+      :nickname,
+      :status,
+      :user,
+      :started,
+      :type,
+      :works_updated,
+      :query,
+      :add,
+      :delete,
+      :replace,
+      :error,
+      :active
+    ])
+    |> unique_constraint(:active)
```

Comment:
You could add `|> validate_inclusion(:status, ["complete", "error", "in_progress"])` for an extra data integrity check here as well